test/Microsoft.ML.Tests/TextLoaderTests.cs

```diff
@@ -1002,6 +1004,126 @@ public void TestWrongDecimalMarkerInputs(bool useCommaAsDecimalMarker)
             }
         }
 
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(false, false)]
+        [InlineData(true, false)]
+        [InlineData(false, true)]
+        public void TestDifferentDecimalMarkersAtTheSameTime(bool useCorrectPeriod, bool useCorrectComma)
+        {
+            // Using 2 different textloaders, with different decimal markers
+            // should yield the expected results even when using their cursors at the same time
+            // in all of the scenarios tested here
+
+            var mlContext = new MLContext(seed: 1);
+
+            var periodPath = GetDataPath("iris.txt");
+            var commaPath = GetDataPath("iris-decimal-marker-as-comma.txt");
+
+            var optionsPeriod = new TextLoader.Options()
+            {
+                Columns = new[]
+            {
```

Comment:
spacing:
```suggestion
                {
```

There's a couple more lines below which could be better aligned.